core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -106,10 +130,29 @@ public void doWork( GHResponse rsp, List<Path> paths, Translation tr )
         {
             String debug = rsp.getDebugInfo() + ", simplify (" + origPoints + "->" + fullPoints.getSize() + ")";
             rsp.setDebugInfo(debug);
+            if (fullPoints.is3D)
+            {
+                int size = fullPoints.getSize();
+                double eleinArr[] = new double[size];
```

Comment:
I would avoid creating a new array here, instead pass `fullPoints` directly to calcAscendDescend

As the smoothing will happen on import (and done in a separate issue). BTW: see also #507 where this then should be placed